File: algorithm/src/main/java/baekjoon/p04435/Main.java
package baekjoon.p04435;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        sc.nextLine();

        for (int tt = 1; tt <= t; tt++) {
            int[][] inputs = new int[2][];
            inputs[0] = intArray(sc.nextLine());
            inputs[1] = intArray(sc.nextLine());
            System.out.println(String.format("Battle %d: %s", tt, solve(inputs)));
        }
    }

    private static int[] intArray(String str) {
        String[] arr = str.split(" ");
        int[] ret = new int[arr.length];

        for (int i = 0; i < ret.length; i++) {
            ret[i] = Integer.parseInt(arr[i]);
        }

        return ret;
    }

    private static String solve(int[][] inputs) {
        int[] a = {1, 2, 3, 3, 4, 10};
        int[] b = {1, 2, 2, 2, 3, 5, 10};

        int aa = mulSum(inputs[0], a);
        int bb = mulSum(inputs[1], b);

        if (aa > bb) {
            return "Good triumphs over Evil";
        } else if (aa < bb) {
            return "Evil eradicates all trace of Good";
        } else {
            return "No victor on this battle field";
        }
    }

    private static int mulSum(int[] a, int[] b) {
        int ret = 0;

        for (int i = 0; i < a.length; i++) {
            ret += (a[i] * b[i]);
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p04435/Main.kt
package baekjoon.p04435

fun main(args: Array<String>) {
    val t = readLine()!!.toInt()
    for (i in 1..t) {
        val inputs = List(2) { readLine()!!.split(" ").map { it.toInt() } }
        println("Battle $i: ${solve(inputs)}")
    }
}

private fun solve(inputs: List<List<Int>>): String {
    val a = arrayOf(1, 2, 3, 3, 4, 10)
    val b = arrayOf(1, 2, 2, 2, 3, 5, 10)

    val aa = (0 until a.size).sumBy { a[it] * inputs[0][it] }
    val bb = (0 until b.size).sumBy { b[it] * inputs[1][it] }

    return when {
        aa > bb -> "Good triumphs over Evil"
        aa < bb -> "Evil eradicates all trace of Good"
        else -> "No victor on this battle field"
    }
}